packages/apps/docs/src/pages/build/guides/election-dapp-tutorial/02-running-devnet.md

```diff
@@ -1,235 +1,305 @@
-transaction that is sent to your local blockchain using the Kadena Client.
-
-In the main function of this file, the `Pact.builder` is used to create a
-transaction for executing the Pact code `(list-modules)`, which is a
-globally available function, not tied to a particular deployed contract.
-Calling this read operation does not cost any gas, so it can be executed
-by passing the transaction to the `dirtyRead` method of the Kadena Client
-instance. Internally, this method transforms the transaction object to a
-json object that is posted to an HTTP API endpoint of your Devnet node.
-The remainder of the main function deals with processing the response
-from the API.
-
-```typescript
-// ./snippets/list-modules.ts
-import { Pact, createClient } from '@kadena/client';
-import { getApiHost, getChainId, getNetworkId } from './configuration';
-
-const client = createClient(getApiHost());
-
-main();
-
-async function main() {
-  const transaction = Pact.builder
-    .execution('(list-modules)')
-    .setMeta({ chainId: getChainId() })
-    .setNetworkId(getNetworkId())
-    .createTransaction();
-
-  try {
-    const response = await client.dirtyRead(transaction);
-
-    const { result } = response;
-
-    if (result.status === 'success') {
-      console.log(result.data);
-    } else {
-      console.error(result.error);
-    }
-  } catch (e: unknown) {
-    console.error((e as Error).message);
-  }
-}
-```
+# Start a local blockchain
+
+In this workshop, you are going to implement the back-end of the election website using  **smart contracts** on the Kadena blockchain. 
+The smart contract you'll be writing for the election application defines rules for:
+
+- Nominating candidates.
+- Casting and counting votes.
+- Storing the nominated candidates and the votes for each candidate.
+  
+Before you publish any smart contract on a public network, like the Kadena test network or the Kadena main network, you should always test that the contract works as expected on your local computer.
+In this tutorial, you'll set up a local development network—`devnet`—to run a blockchain inside of a Docker container on your local computer. 
+You can use this development network to test your smart contracts and experiment with code in an isolated environment that your can reset to a clean state at any time.
+
+## Before you begin
+
+Before you start this tutorial, verify the following basic requirements:
+
+- You have an internet connection and a web browser installed on your local computer.
+- You have a code editor, such as [Visual Studio Code](https://code.visualstudio.com/download), access to an interactive terminal shell, and are generally familiar with using command-line programs.
+- You have cloned the [election-dapp](github.com/kadena-community/voting-dapp.git election-dapp) as described in [Prepare your workspace](/build/guides/election-dapp-tutorial/01-getting-started) and have checkd out the `01-getting-started` branch.
```

Comment:
```suggestion
- You have cloned the [election-dapp](https://github.com/kadena-community/voting-dapp.git) as described in [Prepare your workspace](/build/guides/election-dapp-tutorial/01-getting-started) and have checkd out the `01-getting-started` branch.
```

Reply:
You'll need to update this link in order to make docs build again

Reply:
Thanks for your help, @Ghislain89 !

Reply:
fixed the link along with other changes.